tdms/include/field.h

```diff
@@ -126,6 +126,28 @@ class ElectricSplitField: public SplitField{
      */
     ElectricSplitField(int I_total, int J_total, int K_total) :
             SplitField(I_total, J_total, K_total){};
+
+    /**
+     * @brief Interpolate the x-component to the centre of cell (i,j,k)
+     * 
+     * @param i,j,k Index of the cell to interpolate to the centre of 
+     * @return double Value of the Ex component at the cell centre
+     */
+    double interpolate_x_to_centre(int i, int j, int k);
```

Comment:
Again with my DRY: do we need three different functions `interpolate_{x,y,z}_to_centre` ?

Aaaand now I come to think of it: surely `i, j, k` is a [data clump](https://refactoring.guru/smells/data-clumps) and should be a ... `struct CellCoordinate { int i, ... };` right?

I guess our goal is to have: `Field::interpolate_to_centre_of(CellCoordinate c, Direction::x)` or something.

Reply:
`Field` and `SplitField` are separate classes (don't inherit from a common parent or each other) and the split field is interpolated [here](https://github.com/UCL/TDMS/blob/55e3d910408330513f583097c2bcf2a363a11aaf/tdms/src/iterator.cpp#L1172-L1176) for example. Although functionally they're almost identical as you say: it's just the split field interpolation requires us to add pairs of numbers together prior to actually interpolating.

With regards to moving things up the hierarchy; this is where I'm going in #131, where the `interpolate_x_to_centre` etc are now declared in `Field` and overwritten in `ElectricField` [for example](https://github.com/UCL/TDMS/blob/1a5be0ffc6ee4fc9617e808cb415547a09689b7f/tdms/include/field.h#L209-L340). I'll transfer the relevant changes over to this PR. #80 has been open for a while and is getting quite bloated, so I'm keen to close that to make the diffs smaller and get a concrete base to build off.

On `interpolate_{x,y,z}_to_centre`: now that I know they're working, I'm more open to the idea of making these one function. The ultimate goal of `{Split,}Field::interpolate_to_centre_of(CellCoordinate c, Direction::x)` is quite nice, the trade-off will be a `switch` statement over the `Direction`.

`CellCoordinate` is _definitely_ a data clump - however overhauling that definitely needs it's own separate PR because the `(i,j,k)` usage is so widespread (not just in `iterator.cpp` but across almost all of EG `interpolate.cpp` too). Maybe after #115 it might be more achievable?